LiverMarkups/VTKWidgets/vtkSlicerBezierSurfaceRepresentation3D.cxx

```diff
@@ -193,11 +227,38 @@ void vtkSlicerBezierSurfaceRepresentation3D::UpdateFromMRML(vtkMRMLNode* caller,
 
       this->BezierSurfaceResectionMapper->SetRasToIjkMatrixT(rasToIjkT);
       this->BezierSurfaceResectionMapper->SetIjkToTextureMatrixT(ijkToTextureT);
+      this->BezierSurfaceResectionMapper2D->SetRasToIjkMatrixT(rasToIjkT);
+      this->BezierSurfaceResectionMapper2D->SetIjkToTextureMatrixT(ijkToTextureT);
       }
 
     this->DistanceMapVolumeNode = distanceMap;
     }
 
+  //------------------- add new renderer here ----------------------//
+  auto renderWindow1 = vtkRenderWindow::SafeDownCast(this->GetRenderer()->GetRenderWindow());
+  auto renderers = renderWindow1->GetRenderers();
+
+
+  if(BezierSurfaceDisplayNode->GetShowResection2D())
+    {
+    if(renderers->GetNumberOfItems()!=5)
+      {
+      std::cout<<"-------------------add new renderer------------------"<<endl;
+      double yViewport[4] = {0, 0.6, 0.3, 1.0};
+
+      if (renderWindow1->GetNumberOfLayers() < RENDERER_LAYER+1)
+        {
+        renderWindow1->SetNumberOfLayers( RENDERER_LAYER+1 );
+        }
+      auto CoRenderer2D = vtkSmartPointer<vtkRenderer>::New();
+      CoRenderer2D->SetLayer(RENDERER_LAYER);
+      CoRenderer2D->InteractiveOff();
+      CoRenderer2D->AddActor(this->BezierSurfaceActor2D);
+      CoRenderer2D->SetViewport(yViewport);
+      renderWindow1->AddRenderer(CoRenderer2D);
+      renderWindow1->Render();
```

Comment:
@RuoyanMeng, couldn't we use the viewport manipulation to get the resectogram aspect ratio deformed? In line with one of my previous comments, we could have a resectogram 0-1 in x,y in normalized device coordinates (this will expand the whole viewport). Then we can adjust the width, height of the viewport accordingly? 

@RuoyanMeng Another idea here could be to have a scale widget in the UI to manipulate the viewport, making it smaller or larger. For reference: ALive-research/Slicer-Liver#191